src/Data/LLVM/BitCode.hs

```diff
@@ -1,40 +1,130 @@
+{-# LANGUAGE OverloadedStrings #-}
+
 module Data.LLVM.BitCode (
     -- * Bitcode Parsing
+    -- ** Without 'ParseWarning's
     parseBitCode,     parseBitCodeFromFile
   , parseBitCodeLazy, parseBitCodeLazyFromFile
+    -- ** With 'ParseWarning's
+  , parseBitCodeWithWarnings,     parseBitCodeFromFileWithWarnings
+  , parseBitCodeLazyWithWarnings, parseBitCodeLazyFromFileWithWarnings
 
     -- * Re-exported
   , Error(..), formatError
+  , ParseWarning(..), MetadataRecordSizeRange(..)
+  , ppParseWarnings, ppParseWarning
   ) where
 
 import Data.LLVM.BitCode.Bitstream
     (Bitstream,parseBitCodeBitstream,parseBitCodeBitstreamLazy)
 import Data.LLVM.BitCode.IR (parseModule)
-import Data.LLVM.BitCode.Parse (runParse,badRefError,Error(..),formatError)
+import Data.LLVM.BitCode.Parse (runParse,badRefError,Error(..),formatError,
+                                MetadataRecordSizeRange(..),ParseWarning(..),
+                                ParseState(..),ppParseWarning,ppParseWarnings)
 import Text.LLVM.AST (Module)
 
 import Control.Monad ((<=<))
 import qualified Control.Exception as X
 import qualified Data.ByteString as S
 import qualified Data.ByteString.Lazy as L
+import Data.Foldable (toList)
 
+-- | Parse the contents of an LLVM bitcode file as a strict 'S.ByteString'. If
+-- parsing succeeds, return the parsed 'Module' in a 'Right' value. Otherwise,
+-- return an 'Error' describing what went wrong in a 'Left' value.
```

Comment:
While I certainly appreciate the additional documentation, I don't know if it's worth spelling out how `Either` is used for error handling (i.e., the significance of a `Lerft` vs. a `Right`). That's a pretty fundamental pattern in Haskell APIs. (Comment applies here and elsewhere.) I think this could be
```suggestion
-- | Parse the contents of an LLVM bitcode file as a strict 'S.ByteString'. If
-- parsing succeeds, return the parsed 'Module'. Otherwise,
-- return an 'Error' describing what went wrong.
```
or even just
```suggestion
-- | Parse the contents of an LLVM bitcode file as a strict 'S.ByteString'.
```

Reply:
I've shortened these to avoid mentioning `Left` or `Right`.